mirascope/core/base/_utils/_get_call_args_field_names_and_validate.py

```diff
@@ -0,0 +1,46 @@
+import inspect
+from collections.abc import Callable
+
+from pydantic import BaseModel
+from pydantic.fields import FieldInfo
+
+from mirascope.core.base.from_call_args import FromCallArgs
+
+
+def get_call_args_field_names_and_validate(
```

Comment:
put this at the bottom as it relies on the functions below (for better readability)